examples/gateway_observers.rs

```diff
@@ -17,7 +17,7 @@ pub struct ExampleObserver {}
 // One struct can be an observer of multiple websocketevents, if needed
 impl Observer<GatewayReady> for ExampleObserver {
     // After we subscribe to an event this function is called every time we receive it
-    fn update(&mut self, _data: &GatewayReady) {
```

Comment:
Why downgrade this? This used to be &self but it was deliberately changed, a mutable reference lets the observers do a lot more

Its not needed exactly here, but that makes every observer deriving the trait need to work around being in just a non mutable reference

Reply:
Ah, as I understand it you made it non mutable to not require a mutex, but having to work around only a & ref is worse that just having a mutex imo

Reply:
Excuse me if this is wrong, but since the observer trait itself has an & ref all implementations of it also need that, no?

Reply:
you can make it have a field with a mutex and then locking the mutex of the field in the non mutable update, but that's pretty convoluted

Reply:
observers then are then rather triggering updates on other structs rather than themselves, I guess that works?